src/project/manifest/mod.rs

```diff
@@ -800,12 +801,45 @@ impl<'de> Deserialize<'de> for ProjectManifest {
 
         let toml_manifest = TomlProjectManifest::deserialize(deserializer)?;
 
-        let mut dependencies = HashMap::from_iter([(SpecType::Run, toml_manifest.dependencies)]);
+        // check duplicate dependencies (run, host, build)
+        let mut dependencies = toml_manifest.dependencies.clone();
```

Comment:
Would be nice to beable to reuse this logic to avoid touching bugfixes not being shared.

Reply:
Can you elaborate this please?

Reply:
Possibly put the whole, "insert" and dedup logic into a function that returns a result. Then we can also make a test function for it and reuse it in both the feature and manifest.